pkg/k8sutil/create.go

```diff
@@ -227,3 +230,93 @@ func createNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) erro
 func ListNamespaces(nsclient corev1typed.NamespaceInterface) (*v1.NamespaceList, error) {
 	return nsclient.List(context.TODO(), metav1.ListOptions{})
 }
+
+// ObjectMetadata uniquely identifies any object in the list of YAML manifests.
+type ObjectMetadata struct {
+	Version string
+	Kind    string
+	Name    string
+}
+
+// YAMLToUnstructured accepts a Kubernetes manifest in YAML format and returns an object of type
+// `unstructured.Unstructured`. This object has many methods that can be used by the consumer to
+// extract metadata from the Kubernetes manifest.
+func YAMLToUnstructured(yamlObj []byte) (*unstructured.Unstructured, error) {
+	u := &unstructured.Unstructured{}
+
+	// Decode YAML into `unstructured.Unstructured`.
+	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	if _, _, err := dec.Decode(yamlObj, nil, u); err != nil {
+		return nil, fmt.Errorf("converting config to unstructured.Unstructured: %w", err)
+	}
+
+	return u, nil
+}
+
+// removeYAMLComments converts YAML to JSON and back again, this removes the comments in the YAML
+// and any extra whitespaces spaces.
+func removeYAMLComments(yamlObj []byte) ([]byte, error) {
```

Comment:
Please put unexported methods after exported ones. It makes it easier to read the source file (the deeper you go, the more details you get).